qendpoint-store/src/main/java/com/the_qa_company/qendpoint/store/EndpointTripleSource.java

```diff
@@ -66,46 +97,50 @@ public CloseableIteration<? extends Statement, QueryEvaluationException> getStat
 		Resource newSubj;
 		IRI newPred;
 		Value newObj;
-		long subjectID = this.endpoint.getHdtConverter().subjectToID(resource);
-		long predicateID = this.endpoint.getHdtConverter().predicateToID(iri);
-		long objectID = this.endpoint.getHdtConverter().objectToID(value);
+		long subjectID = this.endpoint.getHdtConverter().subjectToID(subj);
+		long predicateID = this.endpoint.getHdtConverter().predicateToID(pred);
+		long objectID = this.endpoint.getHdtConverter().objectToID(obj);
 
 		if (subjectID == 0 || subjectID == -1) {
-			newSubj = resource;
+			newSubj = subj;
 		} else {
 			newSubj = this.endpoint.getHdtConverter().subjectIdToIRI(subjectID);
 		}
 		if (predicateID == 0 || predicateID == -1) {
-			newPred = iri;
+			newPred = pred;
 		} else {
 			newPred = this.endpoint.getHdtConverter().predicateIdToIRI(predicateID);
 		}
 		if (objectID == 0 || objectID == -1) {
-			newObj = value;
+			newObj = obj;
 		} else {
 			newObj = this.endpoint.getHdtConverter().objectIdToIRI(objectID);
 		}
 
 		logger.debug("SEARCH {} {} {}", newSubj, newPred, newObj);
 
 		// check if we need to search over the delta and if yes, search
-		CloseableIteration<? extends Statement, SailException> repositoryResult;
+		CloseableIteration<? extends Statement> repositoryResult;
 		if (shouldSearchOverNativeStore(subjectID, predicateID, objectID)) {
+			if (statementOrder != null) {
+				throw new UnsupportedOperationException(
+						"Statement ordering is not supported when searching over the native store");
+			}
 			logger.debug("Searching over native store");
 			count++;
 			if (endpoint.isMergeTriggered) {
 				// query both native stores
 				logger.debug("Query both RDF4j stores!");
-				CloseableIteration<? extends Statement, SailException> repositoryResult1 = this.endpointStoreConnection
-						.getConnA_read().getStatements(newSubj, newPred, newObj, false, resources);
-				CloseableIteration<? extends Statement, SailException> repositoryResult2 = this.endpointStoreConnection
-						.getConnB_read().getStatements(newSubj, newPred, newObj, false, resources);
+				CloseableIteration<? extends Statement> repositoryResult1 = this.endpointStoreConnection.getConnA_read()
+						.getStatements(newSubj, newPred, newObj, false, contexts);
+				CloseableIteration<? extends Statement> repositoryResult2 = this.endpointStoreConnection.getConnB_read()
+						.getStatements(newSubj, newPred, newObj, false, contexts);
 				repositoryResult = new CombinedNativeStoreResult(repositoryResult1, repositoryResult2);
```

Comment:
If there is a failure when getting repositoryResult2 the repositoryResult1 would not be closed. I've fixed similar issues in RDF4J and just realised that the same could happen here. I've not tried to fix the code though, but just wanted to point it out now that I saw it.